score-server/src/main/java/bio/overture/score/server/repository/s3/S3DownloadService.java

```diff
@@ -136,21 +139,31 @@ public ObjectSpecification download(
       } else {
         parts = partCalculator.divide(offset, length);
       }
+      if (objectSpec != null) {
+        fillPartUrls(objectKey, parts, objectSpec.isRelocated(), forExternalUse);
 
-      fillPartUrls(objectKey, parts, objectSpec.isRelocated(), forExternalUse);
-
-      val spec =
-          new ObjectSpecification(
-              objectKey.getKey(),
-              objectId,
-              objectId,
-              parts,
-              length,
-              objectSpec.getObjectMd5(),
-              objectSpec.isRelocated());
-
-      return excludeUrls ? removeUrls(spec) : spec;
+        val spec =
+            new ObjectSpecification(
+                objectKey.getKey(),
+                objectId,
+                objectId,
+                parts,
+                length,
+                objectSpec.getObjectMd5(),
+                objectSpec.isRelocated());
 
+        return excludeUrls ? removeUrls(spec) : spec;
+      } else {
+        ObjectMetadata metadata =
+            s3Client.getObjectMetadata(
+                bucketNamingService.getStateBucketName(objectId), objectKey.getKey());
+        fillPartUrls(objectKey, parts, false, forExternalUse);
+        val spec =
+            new ObjectSpecification(
+                objectKey.getKey(), objectId, objectId, parts, length, metadata.getETag(), false);
```

Comment:
`parts` and `length` variables here are provided to us in the request, not information read from the S3 server. They need to be read from object data like the md5 hash (`eTag`) is.